engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoFile.java

```diff
@@ -80,6 +82,13 @@ public ZonedDateTime getLastModifiedTime() throws IOException {
     return ZonedDateTime.ofInstant(truffleFile.getLastModifiedTime().toInstant(), ZoneOffset.UTC);
   }
 
+  @Builtin.Method(name = "posix_permissions_builtin")
+  @Builtin.WrapException(from = IOException.class, to = PolyglotError.class, propagate = true)
+  @Builtin.ReturningGuestObject
+  public Set<PosixFilePermission> getPosixPermissions() throws IOException {
```

Comment:
As discussed yesterday, the `asGuestObject` method may not be available in the future. Can we
```
return new Array(truffleFile.getPosixPermissions().toArray());
```
instead? Working on `TruffleObject` interop shall be preferred over exposing guest Java objects, if possible. Unless we really need the `Set` semantics, I'd go with `Array`.

CCing @hubertp , @kustosz

Reply:
Turns out there is a problem with _layers of Java objects_. Simple `new Array(truffleFile.getPosixPermissions().toArray());` is mixing two layers and creates an unsound data structure:
- `Array` is proper `TruffleObject` ready to flow through Enso language
- `truffleFile.getPosixPermissions()` returns hosted Java objects - those should never appear in the Enso language to the user.

As such please ignore my suggestion. It cannot work as proposed.